### PR TITLE
Modal:  Fix UI mismatch.

### PR DESCRIPTION
**Pull request by ritikamotwani**
Doc: https://docs.google.com/document/d/1pHImZ2YXcG0WxMU5ll1y9zBygyErR2EjR8xu3Y2lgDw/edit

Abstract: https://app.abstract.com/projects/da6f9bdf-2057-45bf-9dda-8dbf0d261eac/branches/46122b74-6115-4a17-ab48-f710f95b1c71/commits/latest/files/DDF7C217-39F9-45D7-8C53-8D8C88D5FC37/layers/CF77B230-AB4B-4283-A850-239DBD7BB86E?collectionId=20a9858f-71ee-42ec-890e-bc196101544e&collectionLayerId=d2d24b3f-eb41-4eca-a03c-d4e35576fd88&mode=build&selected=root-183A0303-1F35-4DBE-A7A1-4350F64CFF39

### CODE REVIEW

File: stories/Display/ModalStory.tsx [hunk 1]
```diff
@@ -8,18 +8,40 @@ import Button from '../../src/General/Button';
 class ModalStory extends React.Component {
   state = {
     visible: false,
+    visibleCenteredModal: false,
+    visibleCenteredModalWithMoreText: false,
   };
 
   handleOpen = () => {
-    this.setState({ visible: true });
+    this.setState({ ...this.state, visible: true });
```
Comment (rosslo) [inline stories/Display/ModalStory.tsx]: Oh for setState, I think we don't need to destructure other states when updating one root property.

**Comment by rosslo** [comment]
Could we also have this example in storybook? 
image

**Comment by rosslo** [comment]
Btw, I think it'd be great if we can add the purpose of Modal on the storybook. Wdyt?

image
image

**Comment by ritikamotwani** [comment]
> Could we also have this example in storybook?
> image

@rosslo , this type isn't implemented yet. For this Gia will add more guidelines and restrictions to the component later, something like:
1. The primary button would be on the top always.
2. Show a warning is more than 2 buttons are used, as the footer height would keep on increasing.
So this one isn't implemented yet.

**Comment by ritikamotwani** [comment]
> Btw, I think it'd be great if we can add the purpose of Modal on the storybook. Wdyt?
> 
> image
> image

Oh I thought the name would make it clear. But yes, good point. Devs who would use it for teh first time might get confused.

File: stories/Display/ModalStory.tsx (hunk 1 above)
Comment (ritikamotwani) [reply stories/Display/ModalStory.tsx]: It's not required. I have a habit of destructuring this in class components haha

**Comment by ritikamotwani** [comment]
@rosslo , updated this.
Thanks!

**Comment by rosslo** [comment]
The storybook looks good now. 👍  

Overall LGTM, other than this eslint warning. Can merge it after you fix it. 🙏 
image

**Pull request merged**